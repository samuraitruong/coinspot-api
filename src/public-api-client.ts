import { BaseClient } from './base-client';

export class PublicApiClient extends BaseClient {
  constructor() {
    super("","", "https://www.coinspot.com.au/pubapi")
  }
  async getLatest() {
    return this.getRequest("latest");
  }
}
