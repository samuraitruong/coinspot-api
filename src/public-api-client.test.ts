import { PublicApiClient } from './public-api-client';
describe("PublicApiClient tests", () => {
  const client = new PublicApiClient();
  it("getLatest test", async () => {
    const latest = await client.getLatest();
    expect(latest).toMatchSnapshot();
  })
})
