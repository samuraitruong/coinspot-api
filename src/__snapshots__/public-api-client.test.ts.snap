// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicApiClient tests getLatest test 1`] = `
Object {
  "prices": Object {
    "ans": Object {
      "ask": "25",
      "bid": "21",
      "last": "22.99",
    },
    "btc": Object {
      "ask": "30100",
      "bid": "30049",
      "last": "30049",
    },
    "doge": Object {
      "ask": "0.0064",
      "bid": "0.005703",
      "last": "0.006",
    },
    "eos": Object {
      "ask": "4.35",
      "bid": "3.8",
      "last": "4.1",
    },
    "eth": Object {
      "ask": "815",
      "bid": "810.77",
      "last": "815",
    },
    "gas": Object {
      "ask": "2.5",
      "bid": "2.12",
      "last": "2.2407",
    },
    "ltc": Object {
      "ask": "153.52",
      "bid": "140",
      "last": "140",
    },
    "powr": Object {
      "ask": "0.152999",
      "bid": "0.1335",
      "last": "0.135",
    },
    "rfox": Object {
      "ask": "0.048195",
      "bid": "0.044999",
      "last": "0.048194",
    },
    "rhoc": Object {
      "ask": "0.037999",
      "bid": "0.028",
      "last": "0.038",
    },
    "str": Object {
      "ask": "0.24999899",
      "bid": "0.21500001",
      "last": "0.24999899",
    },
    "trx": Object {
      "ask": "0.043499",
      "bid": "0.038811",
      "last": "0.040089",
    },
    "xrp": Object {
      "ask": "0.6901",
      "bid": "0.68170103",
      "last": "0.6901",
    },
  },
  "status": "ok",
}
`;
